### ISSUE TITLE
server error (500)

### ISSUE THREAD

**Issue by JChipault** [USGS-WiM/whispers#538]
On 3/14/19 in live beta 10: I went in and created event for 1 dead eagle. it gave me the error 500 message when i hit save, but sent me an email that 200026 had been created. i did not think it had been saved. I then clicked open the 200026 email to view. created same event 4 times, even though i only did it once [she thinks]! Had to delete 200027-200029

On 3/22/19 in live beta 10: When trying to enter a new location to event 200035, got a 500 server error toast message. If she removed the lat/long and hit submit, it committed.

On 4/1 in live beta 10: Went to change the record status of 200034 to "complete" and got this server error: Error. Event not updated. Error message: {"error":"Server Error (500)"}. Same with 200015 a couple hours later. But then was able to mark 200021 as complete. Tried 200034 again and got server error again.

**Comment by JChipault** [comment]
3/22/19 situation mentioned above might be related to GitHub #525

**Comment by BlakeDraper** [comment]
The only way the event could have been created 4 times is if the request was sent 4 times.

**Comment by JChipault** [comment]
4/1/19 issue reported above has been resolved with v13. Thanks!

**Comment by JChipault** [comment]
Not sure this ticket is totally closed yet... out of the 3 reports in this ticket, just the 4/1 report is known to be resolved. We just had another situation of an event committing 4 times yesterday; awaiting details from that user to see if it's similar to what happened on 3/14. Also have user checking on 3/22 report above.

**Comment by JChipault** [comment]
Ok, got details. 200035 is good to go now (report from 3/22) but still getting server error 500 (similar to report from 3/14). User did a great job of documenting her steps yesterday, so here's how we got one event committing 4 times. User filled out create event info the same as she always does except that she selected the "--" option for significance of diagnosis. When she hit save she got the dreaded server error 500. Did NOT get a popup telling her that an event had been created. Because previous struggles seemed to have been related to lat/long, she went back and cleared out her lat/long and then hit save again and she got an error telling her that her lat/long format were not correct (did NOT get a popup telling her that an event had been created). Thinking maybe there was still a ghost space in the lat/long fields, she cleared those out again and then hit save and got the save lat/long in wrong format error message (did NOT get a popup telling her that an event had been created). So then she entered her lat/long back in and selected the "other" option for significance of diagnosis and THEN was told that an event had been successfully created.

So maybe the "--" option in significance of diagnosis is causing the server error to begin with?? And then user isn't being informed that events are being created along the way, just getting error messages so user assumes event has not been committed.

**Comment by BlakeDraper** [comment]
@JChipault I did a little testing and I think you got it exactly right about the "--" option. This was sending `""` (an empty string) instead of `null` which is what it should do. The only place on the entire WHISPers app where the "--" option is still available is on the Basis of Diagnosis and Significance of Diagnosis for Species selects on the species diagnosis dialog, so its not terribly surprising to me we did not discover this problem until April. 

At any rate, I have applied a fix that ensures a `null` is sent when a user blanks out that choice, so this will be fixed in the next release. Re-opening issue until then.
